### PR TITLE
Issue #98 Implement telemetry

### PR DESCRIPTION
**Pull request by gbraad**
This implements a TelemetryRecord that will be send on most of the interaction with the application, like clicking Start, Stop, Delete, Quit, etc.

### CODE REVIEW

File: Forms/SettingsForm.cs [hunk 1]
```diff
@@ -27,8 +27,17 @@ private async void getConfigurationAndResetChanged()
         {
             this.changedConfigs = new Dictionary<string, dynamic>();
             this.configsNeedingUnset = new List<string>();
-            currentConfig = await Task.Run(Tasks.ConfigView);
-            loadConfigurationValues(currentConfig);
+
```
Comment (gbraad) [inline Forms/SettingsForm.cs:30]: Also solved here: https://github.com/code-ready/tray-windows/pull/110

**Comment by gbraad** [comment]
Wil rebase

**Comment by gbraad** [comment]
Will have to check everything, as merging with visual studio's merge editor is a PITA

**Comment by gbraad** [comment]
@anjannath had repushed yesterday. works as expected

File: Helpers/Actions.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        internal const string ClickDelete = "click delete";
+        internal const string ClickOpenConsole = "click open web console";
+        internal const string OpenPreferences = "open preferences";
+        internal const string ApplyPreferences = "apply new preferences";
+        internal const string OpenStatus = "open status";
+        internal const string OpenAbout = "open about";
+        internal const string CopyOCLoginForDeveloper = "copy oc login for developer";
+        internal const string CopyOCLoginForAdmin = "copy oc login for admin";
+        internal const string Quit = "quit application";
+        internal const string ChangeKubernetesContext = "change Kubernetes context";
```
Comment (anjannath) [inline Helpers/Actions.cs:25]: We don't yet have any use for this action, but in the future

File: Helpers/TaskHelpers.cs [hunk 3]
```diff
@@ -9,6 +9,75 @@ namespace CRCTray.Helpers
...
+
+                        return true;
+                    }
+
+                    if (x is APIException) // This we know how to handle.
+                    {
+                        return true;
+                    }
+
+                    if (x is TaskCanceledException)
```
Comment (anjannath) [inline Helpers/TaskHelpers.cs:34]: `TimeoutException` will not occur here?

Comment (gbraad) [reply Helpers/TaskHelpers.cs:34]: It was probably missed, as it is only in `public static async Task<T> TryTaskAndNotify<T, TArgs>` where it includes `Targs`.

Follow-up refactor as it needs to be added in two places.

File: Helpers/Actions.cs (hunk 2 above)
Comment (gbraad) [reply Helpers/Actions.cs:25]: Copied it to stay 'consistent'. It is currently unused, yes.
Let's create an issue after merge that refers to this line.

**Review by anjannath — APPROVED** [review]

**Comment by gbraad** [comment]
Issues created: #113, #114

**Pull request merged**